rs/state_layout/src/state_layout/tests.rs

```diff
@@ -355,6 +355,52 @@ fn test_last_removal_panics_in_debug() {
     });
 }
 
+#[test]
```

Comment:
Double removal could happen during double loading from both the state sync thread and commit_and_certify thread. When state sync puts a checkpoint in the folder, commit_and_certify may also just find it and decide to load it instead of creating one. In this case, loading and verifying the same checkpoint can happen twice.